### ISSUE TITLE
'none' as a ColorExpression result produces an error after #15906

### ISSUE THREAD

**Issue by Nnnes** [openlayers/openlayers#16048]
TL;DR https://openlayers.org/en/v10.0.0/examples/osm-vector-tiles.html is broken

In v9.2.4 and earlier, OpenLayers supported\* `'none'` as the result of a `ColorExpression` in a `FlatStyle`. For example, see the official [v9.2.4 OSM Vector Tiles example](https://openlayers.org/en/v9.2.4/examples/osm-vector-tiles.html) (Ctrl+F `'none'`).  
After #15906, such usages fail during style building with `Error: failed to parse "none" as color`; see [the same OSM Vector Tiles example with OL v10.0.0](https://openlayers.org/en/v10.0.0/examples/osm-vector-tiles.html).

I cannot find anywhere in the current documentation (other than the OSM example) or type definitions that suggests `'none'` could be an acceptable `ColorLike` value. However, the source code appears to contain special handling for `'none'` in a [couple](https://github.com/openlayers/openlayers/blob/6734ed616f53404591894429da9f31c876d5fc42/src/ol/render/canvas/style.js#L329) of [places](https://github.com/openlayers/openlayers/blob/6734ed616f53404591894429da9f31c876d5fc42/src/ol/render/canvas/style.js#L398). If I understand the #15906 changes correctly, these `if`s will always evaluate to `false` and their inner blocks are unreachable in v10.0.0.

Either
* `'none'` should be supported, so the documentation should mention it and some of the code added by 15906 should be edited, or
* `'none'` should be unsupported, so the OSM Vector Tiles example should be edited and any code with special handling can be removed.

I am not familiar enough with this project to recommend one option over the other. My own code has not been affected by this issue; I only noticed it in the example. I have not checked if there are more examples affected by this.

*\*Even in v9.2.4, I think using `'none'` resulted in some unexpected code paths, like `ColorType` not being added to the type [during expression parsing](https://github.com/openlayers/openlayers/blob/a793fdc859874ce347c7bb1480b64c1fe25c5b51/src/ol/expr/expression.js#L298).*

**Comment by mike-000** [comment]
> the documentation should mention it

It is documented for its original intended pupose in `'text-fill-color'` where it allows a no fill, no hit detection override of the default `'#333'`.

**Comment by ahocevar** [comment]
Updated the documentation to mention the `'none'` fill everywhere, not just on `text-fill-color`.
